Hi Cyberbase

IvoNet opened the pull request:
showdown has been updated to 1.0.2 and it has breaking changes. I fixed those in this pull request
One thing is still not right and that is the test.

It seems that the target="_blank" is generated twice in the test but as far as I can see not in actual "live" tests. These I didn't fix.

I also updated the version numbers and added some ignores to the git ignore list

Hope you find the time to merge and create a new version.

Cheerz,

Ivo.

IvoNet committed:
Fixed braking change.
updated to the new showdown 1.0.2

IvoNet committed:
added intellij ignores

IvoNet committed:
Created a build

IvoNet committed:
also fixed the readme
Unit tests still fail but in real life tests it works
You may fix those (double target="_blank" in the expected

IvoNet committed:
also fixed the readme
Unit tests still fail but in real life tests it works
You may fix those (double target="_blank" in the expected

cybercase commented:
Hello IvoNet, thanks for your PR.

I have just finished investigating about the duplicated "target=_blank" in output html.
I think this could be a bug in showdown code that makes this extension (and any other "output extensions") run twice.
I'm going to report this issue shortly.

Let's wait to hear the showdown author.

cybercase commented on .gitignore:
```diff
@@ -26,3 +26,54 @@ node_modules
+
+## Directory-based project format:
+.idea/
+# if you remove the above rule, at least ignore the following:
```
Since we are not removing the above rule, could we remove the following comments?

IvoNet replied:
I used a plugin for IntelliJ to generate these settings I actually never looked at them :-)
I would say leave the *.iml and the .idea/ and remove the rest...

cybercase replied:
Yes, I agree

cybercase commented on package.json:
```diff
@@ -26,22 +26,22 @@
   },
   "devDependencies": {
-    "chai": "^1.10.0",
+    "chai": "^3.0.0",
```
Do we really need to update these devDependencies?

IvoNet replied:
No of course not but I tend to try to stay updated and two major versions seems like a lot...

cybercase replied:
I've checked the Chai changelog and seems that no breaking changes affecting our tests have been introduced in current major version, so It's ok to update.

About the other packages, the "^" operator should do the trick by automatically fetching the latest compatible version. So, in case of non breaking minor version updates, there is no need to update.

cybercase commented on test/cases/basic.html:
```diff
@@ -1,3 +1,3 @@
 
-<p>Link to <a target="_blank" href="http://google.com">Google</a></p>
\ No newline at end of file
+<p>Link to <a target="_blank" target="_blank" href="http://google.com">Google</a></p>
```
Since this broken test could be due to a showdown bug, I would let this line unchanged and and see if the showdown author can fix the bug instead of we changing the test.

IvoNet replied:
I could not commit with failing tests so I cheated :-(

IvoNet commented:
BTW if you do not tag the code Bower will always show the dependency as:

`"showdown-target-blank": "*",`

If you do something like the example below after committing your changes

``` sh
git tag v1.0.0
git push --tags
```

Bower will know a new version because it will look at the tags

if you do that and then in your local repository:

`npm publish ./`

it will also know a version.

Cheerz,

Ivo.

cybercase commented:
Yes, I definitely have to add a tag.
I'm tagging the current version as 0.0.2, so the version after this PR will be the 1.0.0.

Also, I've spoken with the showdown author and solved the bug about the duplicated "target=_blank". He merged my PR and now he just need to update the npm package.

Once the new package is published, we can restore the broken test and merge everything :)

Thanks

Stefano

cybercase commented on package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "showdown-target-blank",
-  "version": "0.0.2",
+  "version": "0.0.3",
```
Since this PR is a breaking change, I think we should make this the 1.0.0.

IvoNet committed:
Fixed test.
Upgraded to showdown 1.1.0
upgraded to the new extension way of working. (eliminated the Deprecated warning)
Added the type thing

IvoNet commented:
I've just updated my version ... Please have a look.
Showdown has upgraded to 1.1.0 and I did to.
I also migrated to the new extension way of working
I hope you can make a release soon because right now I'm working on a "hacked" copy.
